### ISSUE TITLE
eslint_d could not connect

### ISSUE THREAD

**Issue by flashios09** [mantoni/eslint_d.js#61]
Hi,
I installed **eslint_d globally** `npm i eslint_d -g` and tried to lint a file but when i run `eslint_d app.js` nothing happens(more than 2min without anything) while the `eslint app.js` cmd display the lint errors.
I tried `eslint_d status` and i have this message `could not connect`.
![image](https://cloud.githubusercontent.com/assets/1313482/22602612/40873fc8-ea44-11e6-8c75-b66ae424d4b0.png)

I checked the `~/.eslint_d` and the file has one line `53288 59a364a04c2ec7b5`.

The `eslint_d --version` give me this `v3.14.1 (eslint_d v4.2.1)`, the help cmd for `eslint_d` works also(the same output of `eslint --help`).

Is there a log file that i can take a look to understand why eslint_d could not connect ?
Thanks

**Comment by mantoni** [comment]
That is odd. I guess the error occurs here: https://github.com/mantoni/eslint_d.js/blob/master/lib/client.js#L13

Maybe you can add an `err` parameter and add it to the output? That should give us some more information.

**Comment by flashios09** [comment]
@mantoni i create a new file to test `net.connect`
```javascript
    var net = require('net');
    var socket = net.connect(3000, function () {
      console.log('connected');
    });
    socket.on('error', function () {
      console.log('Could not connect');
    });
```
then `node test.js` and it give me `Could not connect`, i uninstalled `nodejs`(version 6.x) and reinstall it(version 7.x) via brew, today i will re-test it

Thanks

**Comment by flashios09** [comment]
> i uninstalled nodejs(version 6.x) and reinstall it(version 7.x)
This solved the problem and now when i do `eslint_d status` i have `running`

Thanks

**Comment by AlexKvazos** [comment]
I had this issue, the fix is to remove `~/.eslint_d`.

I think we need to handle the error that if it fails to connect to the server (because it got killed somehow) it should delete the reference to its port so a new one can start.

**Comment by mantoni** [comment]
@AlexKvazos Sounds like a good idea. Would you care to contribute a pull request with your suggested solution?

**Comment by AlexKvazos** [comment]
PR #95 Sent 👍

**Comment by mantoni** [comment]
A solution is now available in `v7.1.1` (see [release notes](https://github.com/mantoni/eslint_d.js/blob/master/CHANGES.md#711)).
